pyo3-macros/src/lib.rs

```diff
@@ -55,26 +67,152 @@ pub fn pyproto(_: TokenStream, input: TokenStream) -> TokenStream {
     .into()
 }
 
+/// A proc macro used to expose Rust structs as Python objects.
+///
+/// `#[pyclass]` accepts the following [parameters][2]:
+///
+/// |  Parameter  |  Description |
+/// | :-  | :- |
+/// | `name = "python_name"` | Sets the name that Python sees this class as. Defaults to the name of the Rust struct. |
+/// | `freelist = N`  |  Implements a [free list][10] of size N. This can improve performance for types that are often created and deleted in a row. Profile your code to see whether `freelist` is right for you.  |
+/// | `gc`  | Participate in Python's [garbage collection][5]. Required if your type contains references to other Python objects. If you don't (or incorrectly) implement this, contained Python objects may be hidden from Python's garbage collector and you may leak memory. Note that leaking memory, while undesirable, [is safe behavior][7].|
+/// | `weakref` | Allows this class to be [weakly referenceable][6]. |
+/// | `extends = BaseType`  | Use a custom baseclass. Defaults to [`PyAny`][4] |
+/// | `subclass` | Allows Python classes to inherit from this class.  |
+/// | `unsendable`<!-- hack to stop the column from wrapping -->&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160;&#160; | Required if your struct is not [`Send`][3]. Rather than using `unsendable`, consider implementing your struct in a threadsafe way by e.g. substituting [`Rc`][8] with [`Arc`][9]. By using `unsendable`, your class will panic when accessed by another thread.|
+/// | `module = "module_name"` |  Python code will see the class as being defined in this module. Defaults to `builtins`. |
+///
+/// For more on creating Python classes,
+/// see the [class section of the guide][1].
+///
+/// [1]: https://pyo3.rs/main/class.html
+/// [2]: https://pyo3.rs/main/class.html#customizing-the-class
+/// [3]: std::marker::Send
+/// [4]: ../prelude/struct.PyAny.html
+/// [5]: https://pyo3.rs/main/class/protocols.html#garbage-collector-integration
+/// [6]: https://docs.python.org/3/library/weakref.html
+/// [7]: https://doc.rust-lang.org/nomicon/leaking.html
+/// [8]: std::rc::Rc
+/// [9]: std::sync::Arc
+/// [10]: https://en.wikipedia.org/wiki/Free_list
 #[proc_macro_attribute]
 pub fn pyclass(attr: TokenStream, input: TokenStream) -> TokenStream {
     pyclass_impl(attr, input, PyClassMethodsType::Specialization)
 }
 
+/// A proc macro used to expose Rust structs as Python objects.
```

Comment:
Ungh, duplicating this doc kinda hurts, but it's my fault.

For now I think it's okay to copy-paste, I'll try to see if there's a way to make it work nicer in the future.